content/en/blog/_posts/2021-09-07-data-duplication-in-data-heavy-k8s-env.md

```diff
@@ -0,0 +1,221 @@
+
+It’s convenient to create a copy of your application with a copy of its state for each team. 
+For example, you might want a separate database copy to test some significant schema changes 
+or develop other disruptive operations like bulk insert/delete/update...
+
+**Duplicating data takes a lot of time.** That’s because you need first to download 
+all the data from a source like object storage (S3 or Cloud Storage) and then send 
+it back to a storage back-end like EBS or Standard Persistent Disk. There’s a lot of network traffic and CPU/RAM used in this process.
+
+Hardware acceleration by offloading certain expensive operations is **always a huge 
+performance boost**. It reduces the time required to complete an operation by orders 
+of magnitude - for example, GPU computation while you’re watching a high-resolution 
+video on Youtube, mining crypto, or training an ML model, CPU instructions for AES 
+disk encryption or TLS, TCP offload (TOE), and many many more.
+
+## Volume Snapshots to the rescue
+
+Kubernetes introduced Volume Snapshots alpha in 1.12, beta in 1.17, and the Generally Available 
+version in 1.20. Volume Snapshots basically allows Kubernetes to use the cloud-native storage 
+snapshotting mechanism that allows data copy to offload from compute to the actual storage device.
+
+Since data is already in the same storage device (array of devices), duplicating data is usually 
+a metadata operation. All you need to do is point a new disk to an immutable snapshot and only 
+save deltas (or let it do a full-disk copy). As an operation that is inside the storage back-end,
+it’s much quicker and usually doesn’t involve sending traffic over the network.
+
+## Sounds like we have our solution, right?
+
+Actually, Volume Snapshots are namespaced, and Kubernetes protects namespaced data from 
+being shared between tenants (namespaces). This Kubernetes limitation is a conscious design decision so that a pod in a different namespace can’t mount another application’s Persistent Volume (Claim).
+
+One way around it would be to create duplicate data accessible on the same namespace. 
+But if it says that it’s the “production” namespace and many teams are working on this data,
+someone with a fat finger might accidentally swap config files and end up messing up 
+production or data used by other teams.
+
+So the idea is to separate teams/initiatives by namespaces to avoid that and generally 
+limit access to the production namespace.
+
+## Solution? Creating a Golden Snapshot externally
+
+Another way around this design limitation is to create Snapshot externally (not through Kubernetes).
+This is also called pre-provisioning a snapshot manually. Next, I will import it 
+as a multi-tenant golden snapshot that can be used for many namespaces.
+
+High-level plan for preparing the Golden Snapshot:
+1. Identify Disk (EBS/Persistent Disk) that you want to clone with data in the cloud provider
+2. Make a Disk Snapshot (in CSP console)
```

Comment:
either formally introduce this acronym and fully adopt it throughout

> cloud service provider (CSP)

or replace with 

> cloud provider console